src/Lookup.hs

```diff
@@ -90,6 +93,51 @@ multiLookupInternal allowLookupInAllModules name rootEnv = recursiveLookup rootE
           in --(trace $ "multiLookupInternal '" ++ name ++ "' " ++ show (envModuleName env) ++ ", spine: " ++ show (fmap snd spine) ++ ", leaves: " ++ show (fmap snd leaves) ++ ", above: " ++ show (fmap snd above))
             spine ++ leaves ++ above
 
+binderToEnv :: Binder -> Maybe Env
```

Comment:
The new function in this module look nice and general, but since there are no removed code I'm a bit wary of all the code duplication. Many of them seem to be slightly different versions of the ones already there, like `multiLookupALL`, etc. Some more general and some less so. Since you have this fresh in your mind @scolsen, do you agree with my sentiment?

Reply:
Yep that's correct--nearly all of it is just lifted out of the body of `multiLookup`

I agree we should remove the dupes....I didn't remove them in this set of commits since I felt that'd be a potentially tricky change to get right (since we need to make sure the new approach doesn't break `multiLookup`!) So I opted for leaving the duplication in here to keep this set of changes focused on the `implements` stuff only.

I could add a TODO here if that makes sense, unless you think it's OK to go ahead with a complete refactor of Lookups based on the more generic functions.

Reply:
Yeah, I think some TODO:s would be a good solution.

Reply:
added!